### PR TITLE
Extensions: e2e test usePluginComponent hook

### PR DESCRIPTION
**Pull request by sunker**
**What is this feature?**

I'm planning to take on a larger refactoring of the internals in the plugin extensions api (see https://github.com/grafana/grafana/issues/89473). However, the API surface should not be changed. To make this refactoring with confidence, I'd like to have some e2e tests that simply verifies that the `usePluginComponent` works as expected.  

This PR adds a simple app plugin that use the `usePluginComponent` hook to render a component that is exposed by another app plugin. The app plugin that is exposing the component in question is a nested plugin. By testing that the component exposed by plugin A is rendered in plugin B, we know that the entire chain of this api - exposing components and consuming them with the `usePluginComponent` hook - works the way it's intended to. 

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: e2e/plugin-e2e/plugin-e2e-api-tests/as-admin-user/extensions/useExposedComponent.spec.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { test, expect } from '@grafana/plugin-e2e';
+
+const pluginId = 'myorg-componentconsumer-app';
+const exposedComponentTestId = 'exposed-component';
+
+test('should display component exposed by another app', async ({ page }) => {
+  await page.goto(`/a/${pluginId}`);
+  await expect(await page.getByTestId(exposedComponentTestId)).toBeVisible();
+});
```
Comment (leventebalogh) [inline e2e/plugin-e2e/plugin-e2e-api-tests/as-admin-user/extensions/useExposedComponent.spec.ts:9]: Should we also add an `expect()` to check if the passed in props are working correctly? (E.g. to check if `Hello ${name}!` rendered correctly)

**Review by leventebalogh — APPROVED** [review]
Nice! 🚀

File: e2e/plugin-e2e/plugin-e2e-api-tests/as-admin-user/extensions/useExposedComponent.spec.ts (hunk 1 above)
Comment (sunker) [reply e2e/plugin-e2e/plugin-e2e-api-tests/as-admin-user/extensions/useExposedComponent.spec.ts:9]: Yes good shout! I'll fix

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. This is a friendly reminder that the Steady [11.2.x Release Branch](https://github.com/grafana/grafana/tree/v11.2.x) was cut this week, and that it includes the changes committed up to https://github.com/grafana/grafana/commit/b4126d3bce3627e9460e2db0e870c8d984be8e84. This PR was merged after it, so we wanted to let you know that you can add the `backport v11.2.x` label to to backport it and make it part of the next 11.2.x Patch Release, just in case you may have missed the news.

Thank you!

**Pull request merged**